Invoker.hpp

```diff
@@ -1,25 +1,45 @@
+#ifndef INVOKER_HEADER
+#define INVOKER_HEADER
+
+#ifdef _MSC_VER
+#undef __cplusplus
+#define __cplusplus _MSVC_LANG
+#endif
+
+#if __cplusplus > 201402L
 #include <type_traits> 
-#include <concepts>
 #include <functional>
+#if __cplusplus > 201703L
+#include <concepts>
+#endif // __cplusplus > 201703L
 
 template <typename Invocable>
 struct Invoker{
-    using Invocable_T = std::conditional_t<std::is_function_v<Invocable>, 
-    std::add_pointer_t<Invocable>, Invocable>;
-    template <typename I> 
+    using Invocable_T = std::conditional_t<std::is_function_v<Invocable>, std::add_pointer_t<Invocable>, Invocable>;
+#if __cplusplus > 201703L
+    template <typename I>
     requires std::convertible_to<I, Invocable_T>
-    Invoker(I invocable): 
-    invocable_(std::move(invocable)) {} 
+#else // __cplusplus < 202002L
+    template <typename I, typename = typename std::enable_if<std::is_convertible<I, Invocable_T>::value>::type>
+#endif // __cplusplus > 201703L
+    Invoker(I invocable) : invocable_(std::move(invocable)) {}
     
+#if __cplusplus > 201703L
     template <typename... Args>
     requires std::invocable<Invocable&, Args...>
+#else
+    template <typename... Args, typename = typename std::enable_if<std::is_invocable<Invocable_T, Args...>::value>::type>
```

Comment:
please use: 
template <typename... Args, std::enable_if_t<std::is_invocable_v<Invocable_T, Args...>>>

Reply:
actually in both enable_if cases i would use 
static_assert instead in the body of the code 
that way the error messages will be much clearer , enable_if tend to produce a lot of noise

Reply:
in [b6787a5](https://github.com/mguludag/Invoker/commit/b6787a5326f4635dc791a730d0734c31874eaaa8) changed `enable_if `to `static_assert `and include guards to `#pragma once`